StortSpelProjekt/Engine/src/Renderer/SwapChain.h

```diff
@@ -28,12 +28,34 @@ class SwapChain
 	IDXGISwapChain4* GetDX12SwapChain() const;
 	const RenderTargetView* GetRTV(unsigned int backBufferIndex) const;
 	const ShaderResourceView* GetSRV(unsigned int backBufferIndex) const;
+	const bool IsFullscreen() const;
+	void ToggleWindowMode(ID3D12Device5* device,
+		const HWND* hwnd,
+		ID3D12CommandQueue* commandQueue,
+		DescriptorHeap* descriptorHeap_RTV,
+		DescriptorHeap* descriptorHeap_CBV_UAV_SRV);
 
 private:
 	std::array<Resource*, NUM_SWAP_BUFFERS> m_Resources;
 	std::array<RenderTargetView*, NUM_SWAP_BUFFERS> m_RTVs;
 	std::array<ShaderResourceView*, NUM_SWAP_BUFFERS> m_SRVs;
 	IDXGISwapChain4* m_pSwapChain4 = nullptr;
+	DXGI_SWAP_CHAIN_DESC1 scDesc;
+	bool m_Fullscreen = false;
+
+	// colour format
+	DXGI_FORMAT m_DesiredColourFormat = DXGI_FORMAT_R16G16B16A16_FLOAT;
+	unsigned int m_NumberOfSupportedModes = 0;	// the number of supported screen modes for the desired colour format
+	DXGI_MODE_DESC* m_pSupportedModes = {};		// list of all supported screen modes for the desired colour format
+	DXGI_MODE_DESC m_CurrentModeDescription = {};	// description of the currently active screen mode
+
+	const void initFullscreen(unsigned int* width, unsigned int* height);
+	const void resize(unsigned int width, unsigned int height);
+	const void createSwapBuffers(ID3D12Device5* device,
+		unsigned int width, unsigned int height,
+		DescriptorHeap* descriptorHeap_RTV,
+		DescriptorHeap* descriptorHeap_CBV_UAV_SRV);
+	const void clearSwapBuffers() const;
```

Comment:
makes no sense to have this function marked as const because it is supposed to "clearSwapBuffers"? Which means that it should change something inside the class? Also why is the return value marked as const? If the return value is void it wont return anything anyways?

Reply:
Yes, that is a valid point. I do not know why I did that